extensions/wrapper/wrapper-common-api/src/main/java/de/sovity/edc/ext/wrapper/api/common/model/UiPolicyExpression.java

```diff
@@ -0,0 +1,58 @@
+package de.sovity.edc.ext.wrapper.api.common.model;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import io.swagger.v3.oas.annotations.media.Schema;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.RequiredArgsConstructor;
+
+import java.util.List;
+
+
+@Data
+@AllArgsConstructor
+@RequiredArgsConstructor
+@Builder(toBuilder = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@Schema(description = "ODRL constraint as supported by the sovity product landscape")
+public class UiPolicyExpression {
+
+    @Schema(description = "Either LITERAL or one of the constraint types.")
+    private UiPolicyExpressionType expressionType;
+
+    @Schema(description = "Only for types AND, OR, XOR. List of sub-expressions " +
+        "to be evaluated according to the expressionType.")
+    private List<UiPolicyExpression> expressions;
```

Comment:
now that I see that this is infinitely recursive I'm wondering why we still have a List of `UiPolicyExpression` in the UiPolicyCreateRequest instead of one

Reply:
solved